internal/provider/common.go

```diff
@@ -9,30 +10,48 @@ import (
 	"github.com/containerd/containerd/platforms"
 	"github.com/crazy-max/diun/v4/internal/model"
 	"github.com/crazy-max/diun/v4/pkg/registry"
-	"github.com/pkg/errors"
 )
 
 var (
 	metadataKeyChars  = `a-zA-Z0-9_`
 	metadataKeyRegexp = regexp.MustCompile(`^[` + metadataKeyChars + `]+$`)
+	ErrInvalidLabel   = errors.New("invalid label error")
 )
 
+func UpdateMap(m1, m2 map[string]string) map[string]string {
+	if m1 == nil && m2 != nil {
+		return m2
+	}
+
+	if m1 != nil && m2 == nil {
+		return m1
+	}
+
+	if m1 == nil && m2 == nil {
+		return nil
+	}
+
+	for key, value := range m2 {
+		m1[key] = value
+	}
+
+	return m1
+}
+
 // ValidateImage returns a standard image through Docker labels
-func ValidateImage(image string, metadata, labels map[string]string, watchByDef bool) (img model.Image, err error) {
+func ValidateImage(image string, metadata, labels map[string]string, watchByDef bool, imageDefaults model.Image) (img model.Image, err error) {
 	if i := strings.Index(image, "@sha256:"); i > 0 {
 		image = image[:i]
 	}
-	img = model.Image{
-		Name:     image,
-		Metadata: metadata,
-		NotifyOn: model.NotifyOnDefaults,
-		SortTags: registry.SortTagReverse,
-	}
+
+	img = imageDefaults
+	img.Name = image
+	img.Metadata = nil
```

Comment:
Same here you can use `mergo` I think:

```suggestion
	img = model.Image{
		Name: image,
		Metadata: metadata
	}
	if err := mergo.Merge(&img, imageDefaults); err != nil { 
 		return nil, err
	}
```